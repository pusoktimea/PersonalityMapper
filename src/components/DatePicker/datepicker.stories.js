import React from 'react';
import {setAddon, storiesOf} from '@storybook/react';
import {withKnobs, boolean} from '@storybook/addon-knobs';
import JSXAddon from 'storybook-addon-jsx';

import DatePicker from './index';

setAddon(JSXAddon);
const stories = storiesOf('Components', module);

stories.addDecorator(withKnobs);
stories.addWithJSX('Date Picker', () => (
  <DatePicker
    showSummary={boolean('Show Summary', true)}
  />
));
