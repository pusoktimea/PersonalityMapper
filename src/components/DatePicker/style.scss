@import '~compass-mixins/lib/compass';
@import "../../styles/_colors.scss";

$baseClass: '.persmap-date-picker';

#{$baseClass} {
  &_summary {
    margin: 0;
    margin-bottom: 5px;
    color: $doveGray;
    font-weight: 400;
    font-size: 13px;
  }

  &_picker {
    .react-date-picker {
      &__button {
        @include border-radius(5px);
        border: 1px solid $mercury;
        height: 34px;
        &__input {
          padding: 6px 0px 6px 12px;
          
          input {
            font-size: 14px;
            line-height: 1.42857143;
            outline: none;
            height: 100%;
          }
        }
      }
      &__calendar-button {
        @include border-radius(0 5px 5px 0);
        padding: 6px 12px;
        height: 34px;
        border: none;
        border-left: 1px solid $mercury;
        outline: none;
      }
      &__clear-button {
        display: none;
      }
    }

    &_icon {
      font-size: 14px;
      line-height: 1.42857143;
      color: $doveGray;
    }

    &_calendar {
      @include border-radius(5px);
      width: 350px;
      max-width: 100%;
      background: $white;
      border: 1px solid $mercury;
      line-height: 1.125em;
    }
  }
}
