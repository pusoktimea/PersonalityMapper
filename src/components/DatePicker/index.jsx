import React, {Component} from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import moment from 'moment';
import DatePicker from 'react-date-picker';

import Icon from '../Icon';

import './style.scss';

class PersmapDatePicker extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: props.value || new Date()
    };
  }

  onChange = (date) => {
    this.setState({
      value: date
    }, () => {
      if (typeof this.props.onChange === 'function') {
        this.props.onChange(date);
      }
    });
  }

  render() {
    const {value} = this.state;

    const {
      className,
      showSummary
    } = this.props;

    const baseClass = 'persmap-date-picker';
    return (
      <div
        className={cx(
          baseClass,
          className
        )}
      >
        {
          showSummary &&
          <p className={`${baseClass}_summary`}>
            Selected date is: <em>{moment(value).format('dddd, MMMM D, YYYY')}</em>
          </p>
        }
        <DatePicker
          className={`${baseClass}_picker`}
          value={value}
          onChange={this.onChange}
          clearIcon={null}
          calendarIcon={
            <Icon
              className={`${baseClass}_picker_icon`}
              icon="calendar"
            />
          }
          calendarClassName={`${baseClass}_picker_calendar`}
        />
      </div>
    );
  }
}

PersmapDatePicker.propTypes = {
  value: PropTypes.instanceOf(Date),
  onChange: PropTypes.func,
  className: PropTypes.string,
  showSummary: PropTypes.bool
};

PersmapDatePicker.defaultProps = {
  showSummary: true
};

export default PersmapDatePicker;
