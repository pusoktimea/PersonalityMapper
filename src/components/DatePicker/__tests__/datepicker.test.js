import React from 'react';
import renderer from 'react-test-renderer';

import DatePicker from '../index';

describe('DatePicker test', () => {
  it('renders and matches snapshot with summary', () => {
    expect(
      renderer.create(
        <DatePicker
          value={new Date('2017-12-25')}
        />
      ).toJSON()
    ).toMatchSnapshot();
  });

  it('renders and matches snapshot without summary', () => {
    expect(
      renderer.create(
        <DatePicker
          showSummary={false}
          value={new Date('2017-12-25')}
        />
      ).toJSON()
    ).toMatchSnapshot();
  });
});
