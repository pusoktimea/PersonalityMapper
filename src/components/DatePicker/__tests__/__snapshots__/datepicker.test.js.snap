// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker test renders and matches snapshot with summary 1`] = `
<div
  className="activate-date-picker"
>
  <p
    className="activate-date-picker_summary"
  >
    Selected date is: 
    <em>
      Monday, December 25, 2017
    </em>
  </p>
  <div
    className="react-date-picker react-date-picker--closed activate-date-picker_picker"
    onFocus={[Function]}
  >
    <div
      className="react-date-picker__button"
    >
      <div
        className="react-date-picker__button__input"
      >
        <input
          max={null}
          min={null}
          name="date"
          onChange={[Function]}
          onFocus={[Function]}
          required={undefined}
          style={
            Object {
              "left": "-9999px",
              "position": "absolute",
              "top": "-9999px",
              "visibility": "hidden",
            }
          }
          type="date"
          value="2017-12-25"
        />
        <input
          className="react-date-picker__button__input__month"
          max={12}
          min={1}
          name="month"
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="--"
          required={null}
          type="number"
          value={12}
        />
        <span
          className="react-date-picker__button__input__divider"
        >
          /
        </span>
        <input
          className="react-date-picker__button__input__day"
          max={31}
          min={1}
          name="day"
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="--"
          required={null}
          type="number"
          value={25}
        />
        <span
          className="react-date-picker__button__input__divider"
        >
          /
        </span>
        <input
          className="react-date-picker__button__input__year"
          max={null}
          min={1000}
          name="year"
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="--"
          required={null}
          step={1}
          type="number"
          value={2017}
        />
      </div>
      <button
        className="react-date-picker__clear-button react-date-picker__button__icon"
        onClick={[Function]}
        onFocus={[Function]}
        type="button"
      />
      <button
        className="react-date-picker__calendar-button react-date-picker__button__icon"
        onBlur={undefined}
        onClick={[Function]}
        onFocus={[Function]}
        type="button"
      >
        <span
          className="activate-icon fa fa-calendar activate-date-picker_picker_icon"
          id={undefined}
          onClick={undefined}
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`DatePicker test renders and matches snapshot without summary 1`] = `
<div
  className="activate-date-picker"
>
  <div
    className="react-date-picker react-date-picker--closed activate-date-picker_picker"
    onFocus={[Function]}
  >
    <div
      className="react-date-picker__button"
    >
      <div
        className="react-date-picker__button__input"
      >
        <input
          max={null}
          min={null}
          name="date"
          onChange={[Function]}
          onFocus={[Function]}
          required={undefined}
          style={
            Object {
              "left": "-9999px",
              "position": "absolute",
              "top": "-9999px",
              "visibility": "hidden",
            }
          }
          type="date"
          value="2017-12-25"
        />
        <input
          className="react-date-picker__button__input__month"
          max={12}
          min={1}
          name="month"
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="--"
          required={null}
          type="number"
          value={12}
        />
        <span
          className="react-date-picker__button__input__divider"
        >
          /
        </span>
        <input
          className="react-date-picker__button__input__day"
          max={31}
          min={1}
          name="day"
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="--"
          required={null}
          type="number"
          value={25}
        />
        <span
          className="react-date-picker__button__input__divider"
        >
          /
        </span>
        <input
          className="react-date-picker__button__input__year"
          max={null}
          min={1000}
          name="year"
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="--"
          required={null}
          step={1}
          type="number"
          value={2017}
        />
      </div>
      <button
        className="react-date-picker__clear-button react-date-picker__button__icon"
        onClick={[Function]}
        onFocus={[Function]}
        type="button"
      />
      <button
        className="react-date-picker__calendar-button react-date-picker__button__icon"
        onBlur={undefined}
        onClick={[Function]}
        onFocus={[Function]}
        type="button"
      >
        <span
          className="activate-icon fa fa-calendar activate-date-picker_picker_icon"
          id={undefined}
          onClick={undefined}
        />
      </button>
    </div>
  </div>
</div>
`;
